iceoryx_utils/test/moduletests/test_unit_duration.cpp

```diff
@@ -18,218 +18,686 @@
-    result = result - time3;
-    EXPECT_DOUBLE_EQ(result.microSeconds<double>(), 4700000.0);
+TEST(Duration_test, CompareDurationIsNotGreaterThanOther)
+{
+    auto time1 = 1_s + 300_us;
+    auto time2 = 1_s + 200_us;
+    auto time3 = 400_us;
+    auto time4 = 100_us;
+    EXPECT_FALSE(time2 > time1);
+    EXPECT_FALSE(time3 > time1);
+    EXPECT_FALSE(time3 > time2);
+    EXPECT_FALSE(time4 > time3);
+}
 
-    auto time4 = 10_s;
+TEST(Duration_test, CompareDurationIsGreaterThanOrEqualToOther)
+{
+    auto time1 = 1_s + 300_us;
+    auto time2 = 1_s + 200_us;
+    auto time3 = 400_us;
+    auto time4 = 100_us;
+    EXPECT_TRUE(time1 >= time2);
+    EXPECT_TRUE(time1 >= time3);
+    EXPECT_TRUE(time2 >= time3);
+    EXPECT_TRUE(time3 >= time4);
+}
 
-    result = result - time4;
-    EXPECT_EQ(result.milliSeconds<double>(), 0.0);
+TEST(Duration_test, CompareDurationIsNotGreaterThanOrEqualToOther)
+{
+    auto time1 = 1_s + 300_us;
+    auto time2 = 1_s + 200_us;
+    auto time3 = 400_us;
+    auto time4 = 100_us;
+    EXPECT_FALSE(time2 >= time1);
+    EXPECT_FALSE(time3 >= time1);
+    EXPECT_FALSE(time3 >= time2);
+    EXPECT_FALSE(time4 >= time3);
 }
 
-TEST(Duration_test, multiplyDuration)
+// END COMPARISON TESTS
+
+// BEGIN ARITHMETIC TESTS
+
+TEST(Duration_test, AddDuration)
 {
-    auto time1 = 5.2_s;
+    auto time1 = 5_s + 200_ms;
     auto time2 = 200_ms;
-    auto time3 = 300000_us;
+    auto time3 = 700000_us;
 
-    auto result = time1 * time2;
-    EXPECT_NEAR(result.seconds<double>(), 1.04, tenTimesEpsilon);
-    result = result * time3;
-    EXPECT_NEAR(result.seconds<double>(), 0.312, tenTimesEpsilon);
+    EXPECT_THAT(time1.milliSeconds<uint64_t>(), Eq(5200U));
 
-    auto time4 = 10_s;
+    auto result = time1 + time2;
+    EXPECT_EQ(result.milliSeconds<uint64_t>(), 5400U);
 
-    result = result * time4;
-    EXPECT_NEAR(result.seconds<double>(), 3.12, tenTimesEpsilon);
+    result = result + time3;
+    EXPECT_EQ(result.microSeconds<uint64_t>(), 6100000U);
 }
 
-TEST(Duration_test, divideDuration)
+TEST(Duration_test, SubtractDuration)
 {
-    auto time1 = 5.2_s;
+    auto time1 = 6_s - 800_ms;
     auto time2 = 200_ms;
     auto time3 = 300000_us;
 
-    auto result = time1 / time2;
-    EXPECT_NEAR(result.seconds<double>(), 26, tenTimesEpsilon);
-    result = result / time3;
-    EXPECT_NEAR(result.seconds<double>(), 86.666666666666666, tenTimesEpsilon);
+    EXPECT_THAT(time1.milliSeconds<uint64_t>(), Eq(5200U));
+
+    auto result = time1 - time2;
+    EXPECT_EQ(result.milliSeconds<uint64_t>(), 5000U);
+
+    result = result - time3;
+    EXPECT_EQ(result.microSeconds<uint64_t>(), 4700000U);
 
     auto time4 = 10_s;
 
-    result = result / time4;
-    EXPECT_NEAR(result.seconds<double>(), 8.666666666666666, tenTimesEpsilon);
+    result = result - time4;
+    EXPECT_EQ(result.milliSeconds<uint64_t>(), 0U);
 }
 
-TEST(Duration_test, constructFromTimespec)
+TEST(Duration_test, MultiplyDuration)
```

Comment:
We should test:
- multiplication with negative number
- multiplication with 1000000000 and a large duration type so that you run into an overflow